sdk/schemaregistry/schema-registry-avro/samples-dev/withEventHubsConsumerClient.ts

```diff
@@ -0,0 +1,114 @@
+import * as dotenv from "dotenv";
+dotenv.config();
+
+// Set these environment variables or edit the following values
+const endpoint = process.env["SCHEMA_REGISTRY_ENDPOINT"] || "<endpoint>";
+const groupName = process.env["SCHEMA_REGISTRY_GROUP"] || "AzureSdkSampleGroup";
+const connectionString = process.env["EVENTHUB_CONNECTION_STRING"] || "";
+const eventHubName = process.env["EVENTHUB_NAME"] || "";
+const consumerGroup = process.env["CONSUMER_GROUP_NAME"] || "";
+
+// Sample Avro Schema for user with first and last names
+const schemaObject = {
+  type: "record",
+  name: "User",
+  namespace: "com.azure.schemaregistry.samples",
+  fields: [
+    {
+      name: "firstName",
+      type: "string"
+    },
+    {
+      name: "lastName",
+      type: "string"
+    }
+  ]
+};
+
+const schema = JSON.stringify(schemaObject);
+
+// Description of the schema for registration
+const schemaDescription: SchemaDescription = {
+  name: `${schemaObject.namespace}.${schemaObject.name}`,
+  groupName,
+  format: "Avro",
+  definition: schema
+};
+
+export async function main() {
+  // Create a new client
+  const schemaRegistryClient = new SchemaRegistryClient(endpoint, new DefaultAzureCredential());
+
+  // Register the schema. This would generally have been done somewhere else.
+  // You can also skip this step and let `encodeMessageData` automatically register
+  // schemas using autoRegisterSchemas=true, but that is NOT recommended in production.
+  await schemaRegistryClient.registerSchema(schemaDescription);
+
+  // Create a new encoder backed by the client
+  const encoder = new SchemaRegistryAvroEncoder(schemaRegistryClient, { groupName });
+
+  const eventHubConsumerClient = new EventHubConsumerClient(
+    consumerGroup,
+    connectionString,
+    eventHubName
+  );
+
+  const subscription = eventHubConsumerClient.subscribe(
+    {
+      // The callback where you add your code to process incoming events
+      processEvents: async (events, context) => {
+        // Note: It is possible for `events` to be an empty array.
+        // This can happen if there were no new events to receive
+        // in the `maxWaitTimeInSeconds`, which is defaulted to
+        // 60 seconds.
+        // The `maxWaitTimeInSeconds` can be changed by setting
+        // it in the `options` passed to `subscribe()`.
+        for (const event of events) {
+          console.log(
+            `Received event: '${JSON.stringify(event)}' from partition: '${
+              context.partitionId
+            }' and consumer group: '${context.consumerGroup}'`
+          );
+          if (event.contentType !== undefined && event.body) {
+            const contentTypeParts = event.contentType.split("+");
+            if (contentTypeParts[0] === "avro/binary") {
+              const decodedEvent = await encoder.decodeMessageData({
+                contentType: event.contentType,
+                data: Uint8Array.from(Object.values(event.body))
+              });
```

Comment:
To streamline the experience of decoding received messages, I am thinking of adding native Event Hubs received message type as a supported input type in `decodeMessageData`. This could also apply to received message types in other messaging clients.

in JS, we can do that without importing by just duplicating the relevant bits of the native type. For instance, we can define the following input type for [Event Hubs messages](https://github.com/Azure/azure-sdk-for-js/blob/154215e694fbd5302c5457dd6325db6f017ae970/sdk/eventhub/event-hubs/src/eventData.ts#L366):
```ts
interface ReceviedEventHubsMessage {
   body: any,
   contentType?: string
}
```

/cc @JoshLove-msft

Reply:
What about adding some kind of utility method to convert from the different native types to MessageWithMetadata?

Reply:
But actually I like your idea better since it would be less work for the user.